src/components/Card.jsx

```diff
@@ -5,14 +5,19 @@ import { useWishlist } from "../context/wishlistcontext";
 import { useFilter } from '../context/filtercontext';
 import { sortedData } from "../context/filtercontext"
 import { sorted_price_list } from "../reducer/filterreducer";
+import { sorted_category_list } from "../reducer/filterreducer";
+import { useCategory } from '../services/category_api';
+
 
 function Card() {
     const { state, dispatch } = useCart();
     const { cardData } = useProduct();
+    const { category } = useCategory();
     const { currState, wishFunc } = useWishlist();
     const { filterState, filterFunc } = useFilter();
```

Comment:
Suggestion: Be consistent with naming states and dispatch
- In cart context change the name fo state to `cart` and dispatch to `cartFunc`, as you have done the same thing for wishlist and Filter